src/components/icons/interfaces.icons.ts

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-unused-vars */
+
+export type iconType = 'animated' | 'static'
+
+// Interface of the icons objects stored in db:
+export interface IconInterface {
+    name: string,
+    do:string,
+    dont: string,
+    tags: string[],
+    category: string | string[],
+    type: iconType,
+    date: string,
+    hasOutlined?: boolean | undefined,
```

Comment:
`hasOutline?` means it is already optional and can be [undefined](https://rules.sonarsource.com/typescript/RSPEC-4782). We dont need both. Moreover, your schema has this field as required boolean, so we shouldnt make it optional here perhaps?

Reply:
You're right, I've re-checked and made both the interface and schema consistent. 
The mistake was in the schema, `hasOutlined` shouldn't be required, since some icons didn't have this property, I've committed the changes, could you please take a look?